### PR TITLE
node local store

### LINKED ISSUE #10: Local store + Index
Local store for statements and index

Probably SQLite but maybe something like rocksdb

Do we need to consider long-term/large dataset storage optimizations at all yet?

### PR DESCRIPTION
**Pull request by vyzo**
Implements the node Local Store with a relational db, backed by SQLite.
Closes #10 

Summary:
- Refactor mcnode to abstract the statement db; RIP old dumb implementation.
- Implement the StatementDB with a generic SQL database and provide an SQLite backend specialization
- Implement compilation of MCQL queries to SQL for evaluation by the StatementDB
- Implement streaming results with ndjson in the /query API
- Implement deletion: Add DELETE to the MCQL grammar and expose through a /delete api

ISSUE: the gx vendored gogo-protobuf does not marshal/unmarshal oneof, which breaks statement bodies.
Looking for resolution in https://github.com/ipfs/go-ipfs/issues/3214
Fixed in https://github.com/mediachain/concat/pull/21/commits/fa3cc53d969dcd807c3eeaa85c82f1cf8187cff5

### CODE REVIEW

File: mc/query/compile.go [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+func compileNamespaceCriteria(ns string) string {
+	switch {
+	case ns == "*":
+		return ""
+	case ns[len(ns)-1] == '*':
+		pre := ns[:len(ns)-2]
+		return fmt.Sprintf("namespace LIKE '%s%%'", pre)
+	default:
+		return fmt.Sprintf("namespace = '%s'", ns)
```
Comment (yusefnapora) [inline mc/query/compile.go]: Is this safe because we won't be able to get a sql injection payload past the MC query parser?

Comment (vyzo) [inline mc/query/compile.go]: It's safe: the grammar is strict, so it won't get past the parser.

**Comment by yusefnapora** [comment]
Okay, finally made it through the commits 😄 

It's looking good to me; I like the streaming query interface, and you did a great job with the test coverage.  I think we should probably merge this in before it gets any bigger :)

**Comment by vyzo** [comment]
Yeah, I think I am done with it :)

**Pull request merged**